pkg/status/status.go

```diff
@@ -0,0 +1,93 @@
+package status
+
+import (
+	"context"
+	"time"
+
+	metallbv1alpha1 "github.com/metallb/metallb-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ConditionAvailable   = "Available"
+	ConditionProgressing = "Progressing"
+	ConditionDegraded    = "Degraded"
+	ConditionUpgradeable = "Upgradeable"
+)
+
+func UpdateStatus(ctx context.Context, client k8sclient.Client, metallb *metallbv1alpha1.Metallb, condition string, reason string, message string) error {
```

Comment:
You can call this "Update", since status is implicit in the package name.
From the user side, it's gonna be `status.Update`

Reply:
done